components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/status_tray/l10n_util.cc

```diff
@@ -0,0 +1,59 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/status_tray/l10n_util.h"
+
+#include <string>
+#include <vector>
+
+#include "base/no_destructor.h"
+#include "base/notreached.h"
+#include "base/strings/utf_string_conversions.h"
+#include "base/win/atl.h"
+#include "base/win/embedded_i18n/language_selector.h"
+#include "base/win/i18n.h"
+#include "brave/components/brave_vpn/browser/connection/wireguard/win/brave_vpn_wireguard_service/status_tray/resources/wireguard_strings.h"
+
+namespace brave_vpn {
+namespace {
+
+constexpr base::win::i18n::LanguageSelector::LangToOffset
+    kLanguageOffsetPairs[] = {
+#define HANDLE_LANGUAGE(l_, o_) {L## #l_, o_},
+        DO_LANGUAGES
+#undef HANDLE_LANGUAGE
+};
+
+std::wstring GetPreferredLanguage() {
+  std::vector<std::wstring> languages;
+  if (!base::win::i18n::GetUserPreferredUILanguageList(&languages) ||
+      languages.size() == 0) {
+    return L"en-us";
+  }
+
+  return languages[0];
+}
+
+const base::win::i18n::LanguageSelector& GetLanguageSelector() {
+  static base::NoDestructor<base::win::i18n::LanguageSelector> instance(
+      GetPreferredLanguage(), kLanguageOffsetPairs);
+  return *instance;
+}
+
+}  // namespace
+
+std::u16string GetLocalizedString(UINT base_message_id) {
```

Comment:
Sorry, I don't understand why this api is needed. Can't we use `GetStringUTF16()` here?

Reply:
GetStringUTF16 works over `ui::ResourceBundle` which is used only inside chrome. All `generate_embedded_i18n` usages also have similar implementations:
- updater  https://source.chromium.org/chromium/chromium/src/+/main:chrome/updater/win/ui/l10n_util.h;l=6?q=l10n_&sq=&ss=chromium
- installer https://source.chromium.org/chromium/chromium/src/+/main:chrome/installer/util/l10n_string_util.h;l=14?q=l10n_string_util.h&sq=&ss=chromium

I also tried to initialize resource bundle but it did work as expected.

Reply:
Found related docs - https://source.chromium.org/chromium/chromium/src/+/main:docs/updater/design_doc.md;l=272-288